object is listed but no longer valid for dbal4

```
symfony new dbal4-demo --webapp && cd dbal4-demo
```

<pre>bin/console make:entity Post
 <font color="#0000AA">created</font>: src/Entity/Post.php
 <font color="#0000AA">created</font>: src/Repository/PostRepository.php
 
 Entity generated! Now let&apos;s add some fields!
 You can always add more fields later manually or by re-running this command.

 <font color="#00AA00">New property name (press &lt;return&gt; to stop adding fields)</font>:
 &gt; metaData

 <font color="#00AA00">Field type (enter </font><font color="#AA5500">?</font><font color="#00AA00"> to see all types)</font> [<font color="#AA5500">string</font>]:
 &gt; ?

<font color="#00AA00">Main Types</font>
  * <font color="#AA5500">string</font>
  * <font color="#AA5500">text</font>
  * <font color="#AA5500">boolean</font>
  * <font color="#AA5500">integer</font> or <font color="#AA5500">smallint</font> or <font color="#AA5500">bigint</font>
  * <font color="#AA5500">float</font>

<font color="#00AA00">Relationships/Associations</font>
  * <font color="#AA5500">relation</font> or a wizard 🧙 will help you build the relation
  * <font color="#AA5500">ManyToOne</font>
  * <font color="#AA5500">OneToMany</font>
  * <font color="#AA5500">ManyToMany</font>
  * <font color="#AA5500">OneToOne</font>

<font color="#00AA00">Array/Object Types</font>
  * <font color="#AA5500">array</font> or <font color="#AA5500">simple_array</font>
  * <font color="#AA5500">json</font>
  * <font color="#AA5500">object</font>
  * <font color="#AA5500">binary</font>
  * <font color="#AA5500">blob</font>

<font color="#00AA00">Date/Time Types</font>
  * <font color="#AA5500">datetime</font> or <font color="#AA5500">datetime_immutable</font>
  * <font color="#AA5500">datetimetz</font> or <font color="#AA5500">datetimetz_immutable</font>
  * <font color="#AA5500">date</font> or <font color="#AA5500">date_immutable</font>
  * <font color="#AA5500">time</font> or <font color="#AA5500">time_immutable</font>
  * <font color="#AA5500">dateinterval</font>

<font color="#00AA00">Other Types</font>
  * <font color="#AA5500">ascii_string</font>
  * <font color="#AA5500">decimal</font>
  * <font color="#AA5500">guid</font>


 <font color="#00AA00">Field type (enter </font><font color="#AA5500">?</font><font color="#00AA00"> to see all types)</font> [<font color="#AA5500">string</font>]:
 &gt; object

<font color="#00AA00">Main Types</font>
  * <font color="#AA5500">string</font>
  * <font color="#AA5500">text</font>
  * <font color="#AA5500">boolean</font>
  * <font color="#AA5500">integer</font> or <font color="#AA5500">smallint</font> or <font color="#AA5500">bigint</font>
  * <font color="#AA5500">float</font>

<font color="#00AA00">Relationships/Associations</font>
  * <font color="#AA5500">relation</font> or a wizard 🧙 will help you build the relation
  * <font color="#AA5500">ManyToOne</font>
  * <font color="#AA5500">OneToMany</font>
  * <font color="#AA5500">ManyToMany</font>
  * <font color="#AA5500">OneToOne</font>

<font color="#00AA00">Array/Object Types</font>
  * <font color="#AA5500">array</font> or <font color="#AA5500">simple_array</font>
  * <font color="#AA5500">json</font>
  * <font color="#AA5500">object</font>
  * <font color="#AA5500">binary</font>
  * <font color="#AA5500">blob</font>

<font color="#00AA00">Date/Time Types</font>
  * <font color="#AA5500">datetime</font> or <font color="#AA5500">datetime_immutable</font>
  * <font color="#AA5500">datetimetz</font> or <font color="#AA5500">datetimetz_immutable</font>
  * <font color="#AA5500">date</font> or <font color="#AA5500">date_immutable</font>
  * <font color="#AA5500">time</font> or <font color="#AA5500">time_immutable</font>
  * <font color="#AA5500">dateinterval</font>

<font color="#00AA00">Other Types</font>
  * <font color="#AA5500">ascii_string</font>
  * <font color="#AA5500">decimal</font>
  * <font color="#AA5500">guid</font>

<span style="background-color:#AA0000"><font color="#AAAAAA">                                                                                                                        </font></span>
<span style="background-color:#AA0000"><font color="#AAAAAA"> [ERROR] Invalid type &quot;object&quot;.                                                                                         </font></span>
<span style="background-color:#AA0000"><font color="#AAAAAA">                                                                                                                        </font></span>


 <font color="#00AA00">Field type (enter </font><font color="#AA5500">?</font><font color="#00AA00"> to see all types)</font> [<font color="#AA5500">string</font>]:
 &gt; 

</pre>

--- a/src/Maker/MakeEntity.php
+++ b/src/Maker/MakeEntity.php
@@ -430,26 +430,13 @@ final class MakeEntity extends AbstractMaker implements InputAwareMakerInterface
     {
         $allTypes = $this->getTypesMap();
 
-        if ('Hyper' === getenv('TERM_PROGRAM')) {
-            $wizard = 'wizard 🧙';
-        } else {
-            $wizard = '\\' === \DIRECTORY_SEPARATOR ? 'wizard' : 'wizard 🧙';
-        }
-
         $typesTable = [
             'main' => [
-                'string' => [],
+                'string' => ['ascii_string'],
                 'text' => [],
                 'boolean' => [],
                 'integer' => ['smallint', 'bigint'],
                 'float' => [],
-            ],
-            'relation' => [
-                'relation' => 'a '.$wizard.' will help you build the relation',
-                EntityRelation::MANY_TO_ONE => [],
-                EntityRelation::ONE_TO_MANY => [],
-                EntityRelation::MANY_TO_MANY => [],
-                EntityRelation::ONE_TO_ONE => [],
             ],
             'array_object' => [
                 'array' => ['simple_array'],
@@ -469,20 +456,50 @@ final class MakeEntity extends AbstractMaker implements InputAwareMakerInterface
 
         $printSection = static function (array $sectionTypes) use ($io, &$allTypes) {
             foreach ($sectionTypes as $mainType => $subTypes) {
+                if (!\array_key_exists($mainType, $allTypes)) {
+                    // The type is not a valid DBAL Type - don't show it as an option
+                    continue;
+                }
+
+                foreach ($subTypes as $key => $potentialType) {
+                    if (!\array_key_exists($potentialType, $allTypes)) {
+                        // The type is not a valid DBAL Type - don't show it as an "or" option
+                        unset($subTypes[$key]);
+                    }
+
+                    // Remove type as not to show it again in "Other Types"
+                    unset($allTypes[$potentialType]);
+                }
+
+                // Remove type as not to show it again in "Other Types"
                 unset($allTypes[$mainType]);
+
                 $line = sprintf('  * <comment>%s</comment>', $mainType);
 
-                if (\is_string($subTypes) && $subTypes) {
-                    $line .= sprintf(' or %s', $subTypes);
-                } elseif (\is_array($subTypes) && !empty($subTypes)) {
+                if (!empty($subTypes)) {
                     $line .= sprintf(' or %s', implode(' or ', array_map(
                         static fn ($subType) => sprintf('<comment>%s</comment>', $subType), $subTypes))
                     );
-
-                    foreach ($subTypes as $subType) {
-                        unset($allTypes[$subType]);
-                    }
                 }
+
+                $io->writeln($line);
+            }
+
+            $io->writeln('');
+        };
+
+        $printRelationsSection = static function () use ($io) {
+            if ('Hyper' === getenv('TERM_PROGRAM')) {
+                $wizard = 'wizard 🧙';
+            } else {
+                $wizard = '\\' === \DIRECTORY_SEPARATOR ? 'wizard' : 'wizard 🧙';
+            }
+
+            $io->writeln(sprintf('  * <comment>relation</comment> a %s will help you build the relation', $wizard));
+
+            $relations = [EntityRelation::MANY_TO_ONE, EntityRelation::ONE_TO_MANY, EntityRelation::MANY_TO_MANY, EntityRelation::ONE_TO_ONE];
+            foreach ($relations as $relation) {
+                $line = sprintf('  * <comment>%s</comment>', $relation);
 
                 $io->writeln($line);
             }
@@ -494,7 +511,7 @@ final class MakeEntity extends AbstractMaker implements InputAwareMakerInterface
         $printSection($typesTable['main']);
 
         $io->writeln('<info>Relationships/Associations</info>');
-        $printSection($typesTable['relation']);
+        $printRelationsSection();
 
         $io->writeln('<info>Array/Object Types</info>');
         $printSection($typesTable['array_object']);

--- a/tests/Maker/MakeEntityTest.php
+++ b/tests/Maker/MakeEntityTest.php
@@ -74,6 +74,32 @@ class MakeEntityTest extends MakerTestCase
             }),
         ];
 
+        yield 'it_only_shows_supported_types' => [$this->createMakeEntityTest()
+            ->run(function (MakerTestRunner $runner) {
+                $output = $runner->runMaker([
+                    // entity class name
+                    'Developer',
+                    // property name
+                    'keyboards',
+                    // field type
+                    '?',
+                    // use default type
+                    '',
+                    // default length
+                    '',
+                    // nullable
+                    '',
+                    // no more properties
+                    '',
+                ]);
+
+                self::assertStringContainsString('Main Types', $output);
+                self::assertStringContainsString('* string or ascii_string', $output);
+                self::assertStringContainsString('* ManyToOne', $output);
+                self::assertStringNotContainsString('* object', $output);
+            }),
+        ];
+
         yield 'it_creates_a_new_class_and_api_resource' => [$this->createMakeEntityTest()
             ->addExtraDependencies('api')
             ->run(function (MakerTestRunner $runner) {

--- a/tests/Maker/MakeEntityTest.php
+++ b/tests/Maker/MakeEntityTest.php
@@ -96,7 +96,15 @@ class MakeEntityTest extends MakerTestCase
                 self::assertStringContainsString('Main Types', $output);
                 self::assertStringContainsString('* string or ascii_string', $output);
                 self::assertStringContainsString('* ManyToOne', $output);
-                self::assertStringNotContainsString('* object', $output);
+
+                // get the dependencies installed in the test project (tmp/cache/TEST)
+                $installedVersions = require $runner->getPath('vendor/composer/installed.php');
+
+                if (!str_starts_with($installedVersions['versions']['doctrine/dbal']['version'], '3.')) {
+                    self::assertStringNotContainsString('* object', $output);
+                } else {
+                    self::assertStringContainsString('* object', $output);
+                }
             }),
         ];
 

